app/controllers/admin/ambassador_task_assignments_controller.rb

```diff
@@ -0,0 +1,37 @@
+class Admin::AmbassadorTaskAssignmentsController < Admin::BaseController
+  include SortableTable
+  layout "new_admin"
+
+  def index
+    matching_assignments =
+      AmbassadorTaskAssignment
+        .includes(:ambassador_task, ambassador: { memberships: :organization })
+        .completed_assignments(filters: filter_params, sort: { sort_column => sort_direction })
+
+    @ambassador_task_assignments =
+      Kaminari
+        .paginate_array(matching_assignments)
+        .page(params.fetch(:page, 1))
+        .per(params.fetch(:per_page, 25))
+  end
+
+  private
+
+  def sortable_columns
+    %w[completed_at organization_name task_title ambassador_name]
+  end
+
+  def filter_params
+    {}.tap do |h|
```

Comment:
Why use `tap` and not `slice`?

Reply:
Or not just doing the filtering in the query interface in the model?

Reply:
Updated to use `slice`, that's a little more succinct. The query method doesn't need to know that the controller expects filter param keys to be prepended with `search_`

> Connascence of Meaning (CoM) or Connascence of Convention (CoC)
> Connascence of meaning is when multiple components must agree on the meaning of particular values. Returning integers 0 and 1 to represent false and true, respectively, is an example of this form of connascence.
https://en.wikipedia.org/wiki/Connascence#Connascence_of_Meaning_(CoM)_or_Connascence_of_Convention_(CoC)